adapter/internal/api/apis_impl.go

```diff
@@ -173,13 +173,31 @@ func validateAndUpdateXds(apiProject mgw.ProjectAPI, override *bool) (err error)
 		}
 	}()
 
-	// TODO (renuka) when len of apiProject.deployments is 0, return err "nothing deployed" <- check
 	var overrideValue bool
 	if override == nil {
 		overrideValue = false
 	} else {
 		overrideValue = *override
 	}
+
+	// when deployment-environments is missing in the API Project, definition we deploy to default
+	// environment
+	if apiProject.Deployments == nil {
+		conf, _ := config.ReadConfigs()
+		for _, mapping := range conf.Adapter.VhostMapping {
```

Comment:
We should only deploy to the default environment
default env from https://github.com/wso2/product-microgateway/blob/ccf0d7f6b0531a1b625d16266a7e655e24190500/adapter/config/parser.go#L44
default vhost from https://github.com/wso2/product-microgateway/blob/ccf0d7f6b0531a1b625d16266a7e655e24190500/adapter/config/parser.go#L104